builder/azure/common/template/template_builder.go

```diff
@@ -145,6 +145,21 @@ func (s *TemplateBuilder) SetManagedMarketplaceImage(location, publisher, offer,
 	return nil
 }
 
+func (s *TemplateBuilder) SetSharedGalleryImage(location, imageID string) error {
+	resource, err := s.getResourceByType(resourceVirtualMachine)
+	if err != nil {
+		return err
+	}
+
+	s.setVariable("apiVersion", "2018-04-01") // Required for Shared Image Gallery
```

Comment:
Please implement a test that asserts the template is correct when using a shared image.

Reply:
Done